trio/testing/_check_streams.py

```diff
@@ -28,13 +36,13 @@ async def __aexit__(
         traceback: TracebackType | None,
     ) -> None:
         try:
-            await aclose_forcefully(self._both[0])
+            await aclose_forcefully(self._first)
         finally:
-            await aclose_forcefully(self._both[1])
+            await aclose_forcefully(self._second)
 
 
 @contextmanager
-def _assert_raises(exc):
+def _assert_raises(exc: type[BaseException]) -> Generator[None, None, None]:
```

Comment:
Could do `Iterator[None]` here

Reply:
Personally I would keep is annotated as Generator, since it is a generator, I feel like Iterator is best used to describe the return type for `__iter__` on classes

Reply:
Personally I'm a fan of using `Iterable[T]` (not `Iterator`!) when you use `yield` but don't intend people to use [generator-only interfaces like `.send(...)`, `.throw(...)`, or `.close()`](https://docs.python.org/3/reference/expressions.html#generator-iterator-methods).  In this case it doesn't matter though, since we immediately wrap with `@contextmanager`.

Reply:
The contextmanager stubs are actually slightly wrong - a generic iterator won't work, only a generator (since they use send/throw). There's an [open issue in typeshed](https://github.com/python/typeshed/issues/2772) about fixing this, but there's the problem of causing a bunch of nuisance errors in existing code.

Outside this context I do also prefer `Iterable` to avoid promising `send`/`throw`.